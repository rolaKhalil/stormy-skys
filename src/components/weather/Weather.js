import React from 'react'

const Weather = props => (
    <div className="weather__info">
        {
            props.city && <p>
                <img src={'https://www.metaweather.com/static/img/weather/png/64/' + props.weatherStateAbbr + '.png'} alt="Weather"  />
            </p> 
        }
        
        {
            props.city && props.country && <p className="weather__key"> Location: 
            <span className="weather__value"> {props.city}, {props.country}</span>
            </p> 
        }

        {
            props.applicabledDate && <p className="weather__key"> Date:  
                <span className="weather__value">{props.applicabledDate}</span>
            
            </p>

        }

        {
            props.weatherStateName && <p className="weather__key"> Weather: 
                <span className="weather__value">{props.weatherStateName}</span>
            
            </p>

        }

        {
            props.minTemp && <p className="weather__key"> Minimum Temperature: 
                <span className="weather__value">{props.minTemp}</span>
            
            </p>
        
        }

        {
            props.maxTemp && <p className="weather__key"> Maximum Temperature:  
                <span className="weather__value">{props.maxTemp}</span>
            
            </p>
        
        }

        {
            props.theTemp && <p className="weather__key"> Current Temperature: 
                <span className="weather__value">{props.theTemp}</span>
            
            </p>
        
        }

        {
            props.humidity && <p className="weather__key"> Humidity:  
                 <span className="weather__value">{props.humidity}</span>
            
            </p>
        
        }

        {
            props.error && <p className="weather__key">{props.error}</p>
        }

    </div> 
)

export default Weather