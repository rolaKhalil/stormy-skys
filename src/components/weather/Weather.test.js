import React from 'react'
import { shallow, configure } from 'enzyme'
import Weather from './Weather';
import renderer from 'react-test-renderer'

describe('<Weather />', () => {
  it('renders without crashing', () => {
      const tree = renderer.create(
          < Weather weatherStateAbbr={'sn'} city={'Toronto'} country={'Canada'} applicabledDate={'2018-11-19'} weatherStateName={'Light Snow'} minTemp={'-10'} maxTemp={'0'} theTemp={'-2'} humidity={'0'} error={'no error'}/>

      )
      expect(tree.toJSON()).toMatchSnapshot()
  });
})