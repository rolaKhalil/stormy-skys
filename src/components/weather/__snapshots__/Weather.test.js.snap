// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather /> renders without crashing 1`] = `
<div
  className="weather__info"
>
  <p>
    <img
      alt="Weather"
      src="https://www.metaweather.com/static/img/weather/png/64/sn.png"
    />
  </p>
  <p
    className="weather__key"
  >
     Location:
    <span
      className="weather__value"
    >
       
      Toronto
      , 
      Canada
    </span>
  </p>
  <p
    className="weather__key"
  >
     Date:
    <span
      className="weather__value"
    >
      2018-11-19
    </span>
  </p>
  <p
    className="weather__key"
  >
     Weather:
    <span
      className="weather__value"
    >
      Light Snow
    </span>
  </p>
  <p
    className="weather__key"
  >
     Minimum Temperature:
    <span
      className="weather__value"
    >
      -10
    </span>
  </p>
  <p
    className="weather__key"
  >
     Maximum Temperature:
    <span
      className="weather__value"
    >
      0
    </span>
  </p>
  <p
    className="weather__key"
  >
     Current Temperature:
    <span
      className="weather__value"
    >
      -2
    </span>
  </p>
  <p
    className="weather__key"
  >
     Humidity:
    <span
      className="weather__value"
    >
      0
    </span>
  </p>
  <p
    className="weather__key"
  >
    no error
  </p>
</div>
`;
